pkg/storage/bucket/user_bucket_client.go

```diff
@@ -22,16 +23,24 @@ type UserBucketClient struct {
 }
 
 func NewUserBucketClient(userID string, bucket objstore.Bucket) *UserBucketClient {
+	return NewUserBucketClientWithPrefix(userID, bucket, "")
+}
+
+func NewUserBucketClientWithPrefix(userID string, bucket objstore.Bucket, prefix string) *UserBucketClient {
 	return &UserBucketClient{
 		UserBucketReaderClient: UserBucketReaderClient{
 			userID: userID,
 			bucket: bucket,
+			prefix: prefix,
 		},
 		bucket: bucket,
 	}
 }
 
 func (b *UserBucketReaderClient) fullName(name string) string {
+	if b.prefix != "" {
```

Comment:
`return path.Join(b.prefix, b.userID, name)` -- this also skips empty elements

Reply:
Cool, I'll update it to use that function.